tests/test_resolver.py

```diff
@@ -103,12 +106,23 @@ def generate_expected_resolved_deps(
         resolved_packages = user_mapping.lookup_packages(set(user_defined_deps))
         ret.update(resolved_packages)
```

Comment:
I realize this is really a comment on #296, but I'm a little suspicious of _using_ `UserDefinedMapping` to test the correctness of `UserDefinedMapping`. It's easy for a bug in that code to manifest in the same way both in the actual data produced by `resolve_dependencies()` _and_ in the expected data produced here, and thus not trigger a failed test.

In other words (and I noticed this while rebasing #313 across #296), there is currently no mention of `DependenciesMapping.USER_DEFINED` in any of our test code... 🤔

Reply:
True, and I pondered on this before using `UserDefinedMapping` in #296. But `test_resolver.py` is not supposed to test the correctness of `UserDefinedMapping`. It's only meant to test the logic in `resolve_dependencies`. If a bug is to manifest in `UserDefinedMapping`, it should be the job of that class's test to catch it.

I can alternatively switch to `DependenciesMapping.USER_DEFINED` in `test_resolver.py` but then I'll also have to copy a large part of the merge currently happening in `UserDefinedMapping` into `generate_expected_resolved_deps` to construct the user mapping from the mapping file and the custom dictionary mapping.

What do you suggest?

Reply:
Ah, I see. I agree with your analysis that `test_resolver` is supposed to test the _combination_ of resolvers, and not each individual one.

We do have `test_user_defined_mapping__well_formated_input_file__parses_correctly` in `test_packages` that does indeed test `UserDefinedMapping` separately (although it only tests `.import_names` and not `.mappings`, which is why I couldn't find `DependenciesMapping.USER_DEFINED` anywhere).

I suggest we leave this PR as-is on this point, and then I'll add some relevant asserts to `test_user_defined_mapping__well_formated_input_file__parses_correctly` in my #313.

Reply:
Added this in commit 5e5904a on #313.